presto-main/src/main/java/io/prestosql/operator/aggregation/ApproximateLongPercentileAggregations.java

```diff
@@ -100,6 +100,54 @@ public static void weightedInput(@AggregationState DigestAndPercentileState stat
         state.setPercentile(percentile);
     }
 
+    // #758 -- fractional weights
+    @InputFunction
+    public static void weightedInput(@AggregationState DigestAndPercentileState state, @SqlType(StandardTypes.BIGINT) long value, @SqlType(StandardTypes.DOUBLE) double weight, @SqlType(StandardTypes.DOUBLE) double percentile)
```

Comment:
`QuantileDigest.add()` converts `long` -> `double` internally, so instead of duplicating all this logic, let's just change the versions of `weightedInput` that take a `long` and delegate to this one. That will allow us to preserve the BIGINT-based variants of the aggregation that are exposed to users, while simplifying the implementation.